packages/host/tests/integration/components/serialization-test.gts

```diff
@@ -5004,6 +5004,94 @@ module('Integration | serialization', function (hooks) {
       assert.ok(isSaved(hassan), `${hassan.id} is saved`);
       assert.strictEqual(hassan.firstName, 'Hassan');
     });
+
+    test('can resolve relative URLs in linksToMany fields irrespective of location of linked cards', async function (assert) {
+      class Category extends CardDef {
+        @field name = contains(StringField);
+      }
+      class Listing extends CardDef {
+        @field category = linksToMany(Category);
+      }
+
+      await setupIntegrationTestRealm({
+        loader,
+        mockMatrixUtils,
+        contents: {
+          'test-cards.gts': { Category, Listing },
+        },
+      });
+
+      let someCategory = new Category({ name: 'Sample Category' });
+      await saveCard(
+        someCategory,
+        `${testRealmURL}Category/someCategory`,
+        loader,
+      );
+      let listingDoc: LooseSingleCardDocument = {
+        data: {
+          type: 'card',
+          id: `${testRealmURL}listing`,
+          attributes: {},
+          relationships: {
+            'category.0': {
+              links: {
+                self: '../Category/someCategory',
```

Comment:
the fact that this is intentionally a bad ref (`..` vs `.`) is good. it proves we are not paying attention to the links.self when loading the data from the included bucket.